server/src/controllers/game.controller.js

```diff
@@ -358,8 +358,24 @@ class GameController {
   // }
 
   // Get All Games By User
-  // async getAllByUser(req, res) {
-  // }
+  async getAllByUser(req, res, next) {
+    const {userId} = req.params;
+    try{
+      const { data } = await GameRepo.fetchAll();
+      const userGames = data.filter((game) => {
+        return (
+          game.owner.user_id == userId ||
+            (game.opponent && game.opponent.user_id == userId) ||
+            (game.spectators?.length > 0 &&  
+              game.spectators.find((spec) => spec.user_id == userId))
+        );
+      });
+
+      return res.status(200).send(response("fetched user games successfully", userGames));
+    } catch(error){
+      next(`Unable to fetch user games ${error}`);
```

Comment:
check codes above, we've moved back to the custom error class
kindly fix that 
Thanks